apis/v1alpha1/httproute_types.go

```diff
@@ -432,19 +432,39 @@ const (
 // HTTPRequestHeaderFilter defines configuration for the RequestHeaderModifier
 // filter.
 type HTTPRequestHeaderFilter struct {
-	// Add adds the given header (name, value) to the request
+	// Set overwrites the request with the given header (name, value)
 	// before the action.
 	//
 	// Input:
 	//   GET /foo HTTP/1.1
+	//   my-header: foo
 	//
 	// Config:
-	//   add: {"my-header": "foo"}
+	//   set: {"my-header": "bar"}
 	//
 	// Output:
 	//   GET /foo HTTP/1.1
+	//   my-header: bar
+	//
+	// Support: Extended
+	// +optional
+	Set map[string]string `json:"set,omitempty"`
```

Comment:
Is the order of operations defined here? set and add will yield different results depending on order, so we should nail that down.

Reply:
I think it needs another discussion, so perhaps opening another PR is better? The issue seems to be tracked by https://github.com/kubernetes-sigs/service-apis/issues/480.